patterns/Prototype/README.md

```diff
@@ -0,0 +1,112 @@
+# Prototype Design Pattern
+
+
+The Prototype Design Pattern is a creational pattern that enables the creation of new objects by copying an existing object. Prototype allows us to hide the complexity of making new instances from the client. The concept is to copy an existing object rather than create a new instance from scratch, something that may include costly operations. The existing object acts as a prototype and contains the state of the object.
+<br>
+## Prototype Pattern in JavaScript
+
+It promotes the creation of objects without specifying the exact class of object to create. This pattern is particularly useful when you want to create instances of complex objects efficiently.
+<br>
+Below is an example of how to implement and use the prototype pattern in JavaScript.
+
+```javascript
+
+// Prototype object
+const vehiclePrototype = {
+  init(make, model) {
+    this.make = make;
+    this.model = model;
+  },
+  getDetails() {
+    return `${this.make} ${this.model}`;
+  },
+};
+
+// Create new instances using the prototype
+const car1 = Object.create(vehiclePrototype);
+car1.init('Toyota', 'Camry');
+
+const car2 = Object.create(vehiclePrototype);
+car2.init('Honda', 'Civic');
+
+console.log(car1.getDetails()); // Output: 'Toyota Camry'
+console.log(car2.getDetails()); // Output: 'Honda Civic'
+
+```
+
+Below is an explanation of the above code.
+<br>
+
+### 1. Prototype object
+
+```javascript
+
+const vehiclePrototype = {
+  init(make, model) {
+    this.make = make;
+    this.model = model;
+  },
+  getDetails() {
+    return `${this.make} ${this.model}`;
+  },
+};
+
+```
```

Comment:
As above, I don't think you need to copy paste _all_ the code since it's easily accessible in the .js file. Instead, focus your explanation on the more important parts (i.e. the use of `Object.create`).